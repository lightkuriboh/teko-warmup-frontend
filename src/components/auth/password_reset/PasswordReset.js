
import { toast } from 'react-toastify'
import React, { useState } from 'react'
import { Card, Fab } from '@material-ui/core'
import 'react-toastify/dist/ReactToastify.css'
import { TextInput } from 'components/form_elements'
import { password_reset } from 'services/auth_services'

export default function PasswordReset (props) {
  const [ username, setUsername ] = useState('')
  const [ email, setEmail ] = useState('')
  const [ buttonDisabled, setButtonDisabled ] = useState(false)

  function clickSubmit () {
    let query_data = {
      username: username,
      email: email
    }
    setButtonDisabled(true)
    password_reset(query_data,
      (resp) => {
        console.log(resp.data)
        props.history.push('/login')
        toast.success('Success, if the information you provided is correct, check your mail!')
      },
      () => {
        toast.error('Error')
        setButtonDisabled(false)
      }
    )
  }
  return (
    <div>
      <Card style={{margin: '0 auto', width: '35%'}}>
        <div style={{margin: '10px'}}>
          <p>
            RESET PASSWORD
          </p>
        </div>
        <hr style={{color: '#e6e6ff'}}/>
        <div>
          <TextInput onChange={ (e) => setUsername(e.target.value) } placeHolder="Username" focus={ true }/>
          <TextInput onChange={ (e) => setEmail(e.target.value) } placeHolder="Email"/>
        </div>
        <div style={{margin: '10px'}}>
          <Fab
            variant="extended"
            size="medium"
            color="primary"
            aria-label="Add"
            onClick={ clickSubmit }
            disabled={ buttonDisabled }
          >
            RESET
          </Fab>
        </div>
      </Card>
    </div>
  )
}
