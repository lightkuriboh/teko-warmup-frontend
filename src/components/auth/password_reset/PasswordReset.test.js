import React from 'react';
import { shallow } from 'enzyme';
import PasswordReset from './PasswordReset'
describe('Password R', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<PasswordReset/>)
    expect(component).toMatchSnapshot()
  })
})