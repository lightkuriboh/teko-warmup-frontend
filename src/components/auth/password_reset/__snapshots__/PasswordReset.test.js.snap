// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password R should render correctly in "debug" mode 1`] = `
<div>
  <WithStyles(ForwardRef(Card))
    style={
      Object {
        "margin": "0 auto",
        "width": "35%",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      <p>
        RESET PASSWORD
      </p>
    </div>
    <hr
      style={
        Object {
          "color": "#e6e6ff",
        }
      }
    />
    <div>
      <TextInput
        onTextChange={[Function]}
        placeHolder="Username"
      />
      <TextInput
        onTextChange={[Function]}
        placeHolder="Email"
      />
    </div>
    <div
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      <WithStyles(ForwardRef(Fab))
        aria-label="Add"
        color="primary"
        disabled={false}
        onClick={[Function]}
        size="medium"
        variant="extended"
      >
        RESET
      </WithStyles(ForwardRef(Fab))>
    </div>
    <hr
      style={
        Object {
          "color": "#e6e6ff",
        }
      }
    />
    <div>
      message here
    </div>
  </WithStyles(ForwardRef(Card))>
</div>
`;
